qiskit_ibm_provider/utils/backend_converter.py

```diff
@@ -0,0 +1,155 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Converters for migration from IBM Quantum BackendV1 to BackendV2."""
+
+from typing import Any, Dict
+
+from qiskit.transpiler.target import Target, InstructionProperties
+from qiskit.utils.units import apply_prefix
+from qiskit.circuit.library.standard_gates import IGate, SXGate, XGate, CXGate, RZGate
+from qiskit.circuit.parameter import Parameter
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.measure import Measure
+from qiskit.circuit.reset import Reset
+from qiskit.providers.models import (
+    BackendConfiguration,
+    BackendProperties,
+    PulseDefaults,
+)
+
+from ..ibm_qubit_properties import IBMQubitProperties
+
+
+def convert_to_target(
+    configuration: BackendConfiguration,
+    properties: BackendProperties = None,
+    defaults: PulseDefaults = None,
+) -> Target:
+    """Uses configuration, properties and pulse defaults
+    to construct and return Target class.
+    """
+    name_mapping = {
+        "id": IGate(),
+        "sx": SXGate(),
+        "x": XGate(),
+        "cx": CXGate(),
+        "rz": RZGate(Parameter("λ")),
+        "reset": Reset(),
+    }
+    custom_gates = {}
+    target = Target(num_qubits=configuration.n_qubits)
```

Comment:
We should call `qubit_props_dict_from_props` once it returns a list and set `qubit_properties=qubit_properties` here from that list return.